src/server.c

```diff
@@ -1124,10 +1163,41 @@ struct nvnc* nvnc_open(const char* address, uint16_t port)
 poll_start_failure:
 	aml_unref(self->poll_handle);
 failure:
-	close(self->fd);
+	if (self->fd >= 0) {
```

Comment:
Let's move `failure:` below `close(self->fd)` and call it `bind_failure:` instead of having this `if` here. `listen_failure` and `handle_failure` will also need to be added in reverse order above `close()`.